specs/merge/fork-choice.md

```diff
@@ -110,8 +117,12 @@ def on_block(store: Store, signed_block: SignedBeaconBlock) -> None:
 
     # [New in Merge]
     if is_merge_block(pre_state, block.body):
+        # Note: unavailable PoW block(s) may later become available.
+        # Nodes should queue such beacon blocks for later processing.
```

Comment:
This case is a bit more tricky and we should think what would be a better way of specifying it. For the case when CL and EL are in sync with the network, current specification works. But for the case when they are syncing we might want to add more clarity to the spec.

Queueing these blocks could raise a deadlock in an optimistic sync process. Suppose, the optimistic sync is happening and EL uses state sync to catch up with the head and the network passed the transition point a while ago, so the state for transition block is no longer available due to state trie pruning. By queueing beacon blocks at this point, CL will stuck and won't inform EL about the most recent head that is available in the network.

The right behaviour during the sync stage would be as follows:
- process transition block optimistically without terminal block verification
- once fully synced, get back to the transition block and check if it's been finalized or not
- verify terminal block if the transition block hasn't been finalized
- do re-org if the terminal block is invalid 

We also may add this clarity in a separate PR.

Reply:
I consider the optimistic sync process as independent from the consensus specs.

"queuing blocks for later processing" in optimistic beacon chain sync means that you will eventually go back to fully validate them later. The architecture of CL/EL split client is independent of core consensus specs and I want to find the right balance here to not taint the specs with implementation details.

That said, I could see how this is confusing. But I do think that more notes about optimistic processing can/should be included in the Engine API as that is where this implemntation specific architecture design lives